src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/HealthCertificateToolkit.swift

```diff
@@ -0,0 +1,89 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+
+import Foundation
+import Compression
+
+public typealias DecodeHealthCertificateResult = Result<HealthCertificateRepresentations, HealthCertificateDecodingError>
+public typealias FetchProofCertificateResult = Result<Data, ProofCertificateFetchingError>
+
+public enum ProofCertificateFetchingError: Error {
+    case something
+    case general
+}
+
+public enum HealthCertificateDecodingError: Error {
+    case something
+    case general
+}
+
+public protocol HealthCertificateToolkitProtocol {
+
+    func decodeHealthCertificate(base45: String) -> DecodeHealthCertificateResult
+
+    func fetchProofCertificate(for healthCertificates: [HealthCertificateRepresentations], completion: (FetchProofCertificateResult) -> Void)
+}
+
+public struct HealthCertificateToolkit: HealthCertificateToolkitProtocol {
+
+    // MARK: - Protocol HealthCertificateToolkitProtocol
+
+    public func decodeHealthCertificate(base45: String) -> DecodeHealthCertificateResult {
+        .success( .fake())
+    }
+
+    public func fetchProofCertificate(for healthCertificates: [HealthCertificateRepresentations], completion: (FetchProofCertificateResult) -> Void) {
+
+    }
+
+    // MARK: - Internal
+
+    func decodeWithBase45(_ base45String: String) -> Data {
+        return Data()
+    }
+
+    func decompress(_ data: Data) -> Data {
+        return Data()
+    }
+
+    func decodeCOSEPayload(_ data: Data) -> Data {
+        return Data()
+    }
+
+    func decodeJSON(_ data: Data) -> Data {
```

Comment:
semantically, this is either `decodeCBOR` or `extractJSON`. 

`decodeJSON` implies that _JSON_ is the input for the funktion

Reply:
Renamed. I think I will need to refactor a lot of naming, as soon as I get more familiar with the CBOR things.